src/action/autopilot.js

```diff
@@ -48,6 +54,62 @@ class AtplAction {
       this._notification.display({ msg: 'Error toggling autopilot', err });
     }
   }
+
+  /**
+   * Update node scores to get better channels via autopilot.
+   * @return {Promise<undefined>}
+   */
+  async updateNodeScores() {
+    try {
+      await this._setNetwork();
+      const scores = await this._readNodeScores();
+      await this._setNodeScores(scores);
+    } catch (err) {
+      log.error('Updating autopilot scores failed', err);
+    }
+  }
+
+  async _setNetwork() {
+    await this._info.getInfo();
+    if (!this._store.network) {
+      throw new Error('Could not read network');
+    }
+  }
+
+  async _readNodeScores() {
+    const { network, settings } = this._store;
+    try {
+      settings.nodeScores[network] = await this._fetchNodeScores(network);
+      this._db.save();
+    } catch (err) {
+      log.error('Fetching node scores failed', err);
+    }
+    return settings.nodeScores[network];
+  }
+
+  async _fetchNodeScores(network) {
+    const baseUri = 'https://nodes.lightning.computer/availability/v1';
+    const uri = `${baseUri}/btc${network === 'testnet' ? 'testnet' : ''}.json`;
+    const response = checkHttpStatus(await fetch(uri));
+    return this._formatNodesScores((await response.json()).scores);
+  }
+
+  _formatNodesScores(jsonScores) {
+    return jsonScores.reduce((map, node) => {
+      map[node.public_key] = node.score / 100000000.0;
+      return map;
+    }, {});
+  }
+
+  async _setNodeScores(scores) {
+    if (!scores) {
```

Comment:
IIUC because `{}` is truthy, we'll want to change this to `if (scores === {}) {`.

Reply:
scores is `store.settings.nodeScores[network]`. This will be undefined if not set since  `store.settings.nodeScores` is initialized as `{}` now.